src/__tests__/App.test.tsx

```diff
@@ -0,0 +1,115 @@
+    expect(newCardInput).toBeInTheDocument();
+  });
+
+  it("should not be possible to add a flashcard when front card value is empty", async () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    const ctaButton = screen.getByRole("button");
+    fireEvent.click(ctaButton);
+
+    let newCardInput = await screen.findByRole("textbox");
+    fireEvent.change(newCardInput, { target: { value: "" } });
+
+    const nextStepButton = await screen.findByText("Next");
+    fireEvent.click(nextStepButton);
+
+    let saveCardButton = screen.queryByText("Save");
+    expect(saveCardButton).not.toBeInTheDocument();
+
+    fireEvent.change(newCardInput, { target: { value: "This is front" } });
+    fireEvent.click(nextStepButton);
+
+    saveCardButton = await screen.findByText("Save");
+    expect(saveCardButton).toBeInTheDocument();
+  });
+
+  it("should not be possible to add a flashcard when back card value is empty", async () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    const ctaButton = screen.getByRole("button");
+    fireEvent.click(ctaButton);
+
+    let newCardInput = await screen.findByRole("textbox");
+    fireEvent.change(newCardInput, { target: { value: "" } });
+
+    const nextStepButton = await screen.findByText("Next");
+    fireEvent.click(nextStepButton);
+
+    let saveCardButton = screen.queryByText("Save");
+    expect(saveCardButton).not.toBeInTheDocument();
+
+    fireEvent.change(newCardInput, { target: { value: "This is front" } });
+    fireEvent.click(nextStepButton);
+
+    saveCardButton = await screen.findByText("Save");
+    expect(saveCardButton).toBeInTheDocument();
+
+    newCardInput = await screen.findByRole("textbox");
+
+    fireEvent.change(newCardInput, { target: { value: "" } });
+    fireEvent.click(saveCardButton);
+    expect(saveCardButton).toBeInTheDocument();
+  });
+
+  it("should be possible to add a flashcard when front and back card value is not empty", async () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    const ctaButton = screen.getByRole("button");
+    fireEvent.click(ctaButton);
+
+    let newCardInput = await screen.findByRole("textbox");
+    fireEvent.change(newCardInput, { target: { value: "" } });
+
+    const nextStepButton = await screen.findByText("Next");
+    fireEvent.click(nextStepButton);
+
+    let saveCardButton = screen.queryByText("Save");
+    expect(saveCardButton).not.toBeInTheDocument();
+
+    fireEvent.change(newCardInput, { target: { value: "This is front" } });
+    fireEvent.click(nextStepButton);
+
+    saveCardButton = await screen.findByText("Save");
+    expect(saveCardButton).toBeInTheDocument();
+
+    newCardInput = await screen.findByRole("textbox");
+
+    fireEvent.change(newCardInput, { target: { value: "" } });
+    fireEvent.click(saveCardButton);
+    expect(saveCardButton).toBeInTheDocument();
```

Comment:
**nitpick:** You don't need this check. You already clicked on this save button one line above

Reply:
this is intended to check whether sending an empty input will save the card and close the form. While writing this comment, I already see that this is not the best solution and I should rather focus on checking the closing of the form, which makes me think whether I should test adding a card by sending a request to the backend and re-rendering the cards to check whether the new card will appear on the list?

Reply:
Yea - you should check if clicking Save actually adds card to the list. Request is being sent already when you click on this button so you need to mock it (I think you did it already?)

Reply:
still have not found a way to mock it.  The test from [PR #17 ](https://github.com/meczajkowski/fischkapp/pull/17) works but i think this is not the way it should be done.